test/index.test.ts

```diff
@@ -257,6 +257,71 @@ describe("mkdist", () => {
     `);
   }, 50_000);
 
+  it("mkdist (emit sourcemaps)", async () => {
+    const rootDir = resolve(__dirname, "fixture");
+    const { writtenFiles } = await mkdist({
+      rootDir,
+      esbuild: {
+        sourcemap: "linked",
+      },
+    });
+
+    expect(writtenFiles.sort()).toEqual(
+      [
+        "dist/README.md",
+        "dist/bar.mjs",
+        "dist/bar.mjs.map",
+        "dist/demo.css",
+        "dist/dir-export.mjs",
+        "dist/dir-export.mjs.map",
+        "dist/foo.mjs",
+        "dist/foo.d.ts",
+        "dist/index.mjs",
+        "dist/index.mjs.map",
+        "dist/types.d.ts",
+        "dist/star/index.mjs",
+        "dist/star/index.mjs.map",
+        "dist/star/other.mjs",
+        "dist/star/other.mjs.map",
+        "dist/components/index.mjs",
+        "dist/components/index.mjs.map",
+        "dist/components/blank.vue",
+        "dist/components/js.vue",
+        "dist/components/script-multi-block.vue",
+        "dist/components/script-multi-block.vue.mjs.map",
+        "dist/components/script-setup-ts.vue",
+        "dist/components/script-setup-ts.vue.mjs.map",
+        "dist/components/ts.vue",
+        "dist/components/ts.vue.mjs.map",
```

Comment:
I'm unsure about the generation of the `*.vue.mjs.map` files, as I haven't worked with `.vue` files before.